gkbuild: Install the correct binaries with slibtool

orbea opened the pull request:
When using genkernel with `sys-devel/slibtool` rather than `sys-devel/libtool` some of the build scripts will install slibtool wrapper scripts instead of the correct binary executable files. This happens because they are installed manually rather than with the `install` target that correctly calls `$LIBTOOL`. This will then fail when `strip(1)` doesn't accept text files as input.

There is already a solution for this in `x11-apps/mesa-progs` in the gentoo repo which I adapted for `util-linux` here.

https://gitweb.gentoo.org/repo/gentoo.git/tree/x11-apps/mesa-progs/mesa-progs-9999.ebuild?id=cd699fd652dc322e78d74db341f56f9b6a3abd11#n65

This might need to be done in more places, but I decided to start with `util-linux` and `cryptsetup` which are failing for me currently.

Additionally to reproduce this issue genkernel needs to use slibtool instead of libtool which is not straight forward. To accomplish this I used the `--utils-make="$HOME"/bin/tools/make` argument for genkernel where `$HOME"/bin/tools/make` is the following short script.
```
#!/bin/sh

export MAKE='make LIBTOOL=rlibtool'
export MAKEFLAGS='LIBTOOL=rlibtool'

make "$@"
```

orbea committed:
gkbuild/util-linux: Install the correct binaries with slibtool

Signed-off-by: orbea <orbea@riseup.net>

orbea committed:
gkbuild/cryptsetup: Install the correct binaries with slibtool

Signed-off-by: orbea <orbea@riseup.net>

robbat2 commented:
@orbea Can you refactor this to be a helper function and use that?
I think there's more places that need the same fix, and refactoring will make it a lot easier

orbea commented:
Sure, I was thinking about that as well when I looked this earlier today. I'll update this soon with the change.

orbea committed:
gen_funcs: Add an install_exe function

This helps when the build uses slibtool where the compiled binary output
path differs from GNU libtool.

Signed-off-by: orbea <orbea@riseup.net>

orbea commented:
@robbat2 Rebased and force pushed with a new commit that adds an `install_exe` function to `gen_funcs.sh`. This is indeed cleaner looking.
